File: src/test/java/net/haesleinhuepf/clij/ops/examples/Mean.java

package net.haesleinhuepf.clij.ops.examples;

import java.io.IOException;

import org.junit.Test;

import net.haesleinhuepf.clij.ops.CLIJ_close.CLIJ_close;
import net.haesleinhuepf.clij.ops.CLIJ_pull.CLIJ_pull;
import net.haesleinhuepf.clij.ops.CLIJ_push.CLIJ_push;
import net.imagej.ImageJ;

public class Mean {

	@Test
	public void run() throws IOException {
		ImageJ ij = new ImageJ();
		Object input = ij.io().open(getClass().getResource("/samples/t1-head.tif").getPath());
		ij.ui().show("input", input);
		// push image to GPU
		Object inputGPU = ij.op().run(CLIJ_push.class, input);

		Object mean = ij.op().run("CLIJ_meanSphere", inputGPU, 3, 3, 3);

		Object result = ij.op().run(CLIJ_pull.class, mean);
		ij.ui().show("mean", result);

		// cleanup
		ij.op().run(CLIJ_close.class, inputGPU);
		ij.op().run(CLIJ_close.class, mean);
		ij.op().run(CLIJ_close.class);

	}

	public static void main(String... args) throws IOException {
		Mean task = new Mean();
		task.run();
	}
}

File: src/test/resources/jython/mean.py
# CLIJ example ImageJ Ops jython: mean.py
#
# This script shows how the mean average filter works in the GPU.
#
# Author: Robert Haase, rhaase@mpi-cbg.de
# Author: Deborah Schmidt, frauzufall@mpi-cbg.de
# May 2019
# ---------------------------------------------

#@ IOService io
#@ UIService ui
#@ OpService ops

import os
import inspect
def getResource(file):
    return os.path.dirname(os.path.abspath(inspect.getsourcefile(lambda:0))) + "/" + file


input = io.open(getResource("../samples/t1-head.tif"))
ui.show("input", input)

# push image to GPU
inputGPU = ops.run("CLIJ_push", input)

# Local mean filter in GPU
mean = ops.run("CLIJ_meanSphere", inputGPU, 3, 3, 3);
# show result
result = ops.run("CLIJ_pull", mean)
ui.show("mean", result)

#cleanup
ops.run("CLIJ_close", inputGPU)
ops.run("CLIJ_close", mean)

